packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -1495,31 +1469,41 @@ void ArbCore::operator()() {
                 }
             }
 
-            if (perform_age_pruning) {
-                // Prune checkpoints that are too old
-                auto prune_status = pruneToTimestampOrBefore(
-                    checkpoint_pruning_age_timestamp,
-                    coreConfig.checkpoint_max_to_prune);
-                if (!prune_status.ok() && !prune_status.IsNotFound()) {
-                    // Non-fatal error
-                    std::cerr << "Error pruning checkpoints: "
-                              << prune_status.ToString() << "\n";
+            if (perform_pruning) {
+                if (coreConfig.checkpoint_pruning_age_seconds > 0) {
+                    // Prune checkpoints that are too old
+                    auto checkpoint_pruning_age_timestamp =
+                        seconds_since_epoch() -
+                        coreConfig.checkpoint_pruning_age_seconds;
+                    auto prune_status = pruneToTimestampOrBefore(
+                        checkpoint_pruning_age_timestamp,
+                        coreConfig.checkpoint_max_to_prune);
+                    if (!prune_status.ok() && !prune_status.IsNotFound()) {
+                        // Non-fatal error
+                        std::cerr << "Error pruning checkpoints: "
+                                  << prune_status.ToString() << "\n";
+                    }
+                }
+
+                uint256_t checkpoint_pruning_gas_used = 0;
+                {
+                    std::lock_guard<std::mutex> lock(checkpoint_pruning_mutex);
+                    checkpoint_pruning_gas_used =
+                        unsafe_checkpoint_pruning_gas_used;
                 }
-                perform_age_pruning = false;
-            }
 
-            if (perform_gas_pruning) {
                 // Prune checkpoints that have used less gas
                 // than specified
-                auto prune_status =
-                    pruneToGasOrBefore(checkpoint_pruning_gas_used,
-                                       coreConfig.checkpoint_max_to_prune);
-                if (!prune_status.ok() && !prune_status.IsNotFound()) {
-                    // Non-fatal error
-                    std::cerr << "Error pruning checkpoints: "
-                              << prune_status.ToString() << "\n";
+                if (checkpoint_pruning_gas_used > 0) {
+                    auto prune_status =
+                        pruneToGasOrBefore(checkpoint_pruning_gas_used,
+                                           coreConfig.checkpoint_max_to_prune);
+                    if (!prune_status.ok() && !prune_status.IsNotFound()) {
+                        // Non-fatal error
+                        std::cerr << "Error pruning checkpoints: "
+                                  << prune_status.ToString() << "\n";
+                    }
                 }
```

Comment:
Should set `perform_pruning` to false here

Reply:
:+1: